sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java

```diff
@@ -574,14 +644,20 @@ public ServiceBusReceiverAsyncClient buildAsyncClient() {
 
             return new ServiceBusReceiverAsyncClient(connectionProcessor.getFullyQualifiedNamespace(), entityPath,
                 entityType, false, receiveMessageOptions, connectionProcessor, tracerProvider,
-                messageSerializer, messageLockContainer);
+                messageSerializer, messageLockContainer, ServiceBusClientBuilder.this::onClientClose);
         }
 
         /**
-         * Creates <b>synchronous</b> Service Bus receiver responsible for reading {@link ServiceBusMessage
-         * messages} from a specific queue or topic.
+         * Creates <b>synchronous</b> Service Bus receiver responsible for reading {@link ServiceBusMessage messages}
+         * from a specific queue or topic.
          *
          * @return An new {@link ServiceBusReceiverClient} that receives messages from a queue or topic.
+         * @throws IllegalStateException if {@link #queueName(String) queueName} or {@link #topicName(String)
+         *     topicName} are not set or, both of these fields are set. It is also thrown if the Service Bus {@link
+         *     #connectionString(String) connectionString} contains an {@code EntityPath} that does not match one set in
+         *     {@link #queueName(String) queueName} or {@link #topicName(String) topicName}. Lastly, if a {@link
+         *     #topicName(String) topicName} is set, but {@link #subscriptionName(String) subscriptionName} is not.
+         * @throws IllegalArgumentException if the entity type is not a queue or a topic.
```

Comment:
Line 660: `IllegalArgumentException if the entity type is not a queue or a topic` : You have already explained all the scenario above this. `UNKNOWN` , type is internal to us and if we can avoid using `entity type` , one less thing to learn for user .   
You do throw this exception but this will only happen if user has not set both queue ,topic and connection string has entityName in it.
Or it may say something like `Queue or topic name are not set via queueNme() or topicName() respectively.`

Reply:
* I added the IllegalArgumentException because checkstyles complained about missing javadocs for the exception.
* I updated the text.